content/posts/2024-11-05-flang-new.md

```diff
@@ -0,0 +1,444 @@
+* Support Fortran tool development by being based on existing LLVM frameworks.
+* Support Fortran language experimentation for future language standards proposals.
+
+The last one can be explained by comparison to Clang. For C++ feature, a common
+path is to implement them on top of Clang, GCC, or both, to prove the feature can work.
+Implementing the feature multiple times in different compilers uncovers any
+assumptions in the design that may be a problem for one compiler or another.
+
+In the same way, Flang and gfortran can provide that diversity.
+
+Finally, though Fortran is an ISO standard, standards can be ambiguous
+and implementations can make mistakes. A new compiler is a chance to
+rethink those decisions and find parts of the standards that have been neglected.
+
+Jeff Hammond (NVIDIA) had this to say on the topic:
+
+TODO: quote goes here
+
+# PGI to LLVM: the Flang Story
+
+Though the first glimpses of Flang as we know it do not appear until 2017, the
+Portland Group's history with supercomputing and US National Labs is important
+so we will start there.
+
+* **1989** The Portland Group, otherwise known as
+  [PGI](https://web.archive.org/web/19970628161656/http://www.pgroup.com/corp_home.html),
+  is formed. To provide C, Fortran 77 and C++ compilers for the Intel i860 market.
+* **1990** Intel bundles PGI's compilers with its iPSC/860 supercomputer.
+* **1996** PGI
+  [works with](https://web.archive.org/web/20100528051556/http://www.sandia.gov/ASCI/Red/papers/Mattson/OVERVIEW.html)
+  Sandia National Laboratories to provide compilers for the ASCI Option Red
+  supercomputer.
+* **December 2000** PGI becomes a
+  [wholly owned subsidiary](https://www.electronicsweekly.com/news/archived/resources-archived/stmicroelectronics-to-acquire-pgi-2000-12/) of
+  STMicroElectronics.
+* **July 2013** PGI is [sold to NVIDIA](https://www.theregister.com/2013/07/30/nvidia_buys_the_portland_group/).
+
+In late 2015 we saw the first signs of what would become "Classic Flang".
+
+* **November 2015** NVIDIA is part of the US Department of Energy
+  Exascale Computing Project. Including a commitment to create an open source
+  [Fortran compiler](https://www.llnl.gov/article/41756/nnsa-national-labs-team-nvidia-develop-open-source-fortran-compiler-technology).
+
+  > “The U.S. Department of Energy’s National Nuclear Security Administration and its
+  > three national labs have reached an agreement with NVIDIA's PGI division to adapt
+  > and open-source PGI's Fortran frontend, and associated Fortran runtime library,
+  > for contribution to the LLVM project.”
+* **May 2017** The first source code release of Classic Flang as a separate
+  respository, outside of the LLVM Project. Consisting of PGI's compiler frontend
+  and a new backend that generated LLVM IR.
+
+  Other companies such as Arm and AMD got involved at this point.
+
+During this time, plans were being made to propose Classic Flang for inclusion
+in the LLVM Project.
```

Comment:
You may want to explain somewhere that _today_ we make the distinction into "Classic Flang" and "LLVM Flang", but back then it was just ... Flang.

Reply:
I've noted this in the prior section where I first use the term, and that I will use classic flang even when classic flang wasn't a term yet, for clarity's sake.